cmake/kokkos_tribits.cmake

```
@@ -6,12 +6,23 @@ include(GNUInstallDirs)
 
 message(STATUS "The project name is: ${PROJECT_NAME}")
 
-if(GTest_FOUND)
-  set(KOKKOS_GTEST_LIB GTest::gtest)
-  message(STATUS "Using gtest found in ${GTest_DIR}")
-else() # fallback to internal gtest
-  set(KOKKOS_GTEST_LIB kokkos_gtest)
-  message(STATUS "Using internal gtest for testing")
+if(Kokkos_ENABLE_TESTS)
+  find_package(GTest QUIET 1.14.0)
+  if(GTest_FOUND)
+    message(STATUS "Found external GoogleTest: ${GTest_DIR} (version \"${GTest_VERSION}\")")
+  else()
+    include(FetchContent)
+    list(APPEND CMAKE_MESSAGE_INDENT "[googletest] ")
+    FetchContent_Declare(googletest SOURCE_DIR ${Kokkos_SOURCE_DIR}/tpls/googletest-1.14.0)
+    FetchContent_MakeAvailable(googletest)
+    list(POP_BACK CMAKE_MESSAGE_INDENT)
+    message(STATUS "Using bundled GoogleTest copy (version \"1.14.0\")")
+
+    # Suppress clang-tidy diagnostics on code that we do not have control over
+    if(CMAKE_CXX_CLANG_TIDY)
+      set_target_properties(gtest gtest_main gmock gmock_main PROPERTIES CXX_CLANG_TIDY "")
+    endif()
```

You might want to also suppress warnings for these targets (we do this already for google benchmark)
```suggestion
    if(CMAKE_CXX_CLANG_TIDY)
      set_target_properties(gtest gtest_main gmock gmock_main PROPERTIES CXX_CLANG_TIDY "")
    endif()
    
    set_target_properties(gtest gtest_main gmock gmock_main PROPERTIES COMPILE_OPTIONS -w)
```

---

Applied.  Thanks